### PR TITLE
Rich Text Editor | Add label provider in nimble-rich-text-editor

### PR DESCRIPTION
**Pull request by aagash-ni**
# Pull Request

## 🤨 Rationale
Issue Link : https://github.com/ni/nimble/issues/1288

Currently, `rich-text-editor` has `title` and `text content` for formatting buttons (Bold, Italics, Bulleted List, Numbered List) and this needs to be localized from the client side.

## 👩‍💻 Implementation

Reference PR for implementation: https://github.com/ni/nimble/pull/1328

This PR  creates label DesignTokens in `nimble-rich-text-editor`

Note: Angular Integration will be taken up once after [this ](https://github.com/ni/nimble/pull/1450)PR completion.

## 🧪 Testing

- Verified that each label appears in the expected spots in the DOM for the affected components. Note that the current labels are mostly not user-visible. Future format button additions will introduce new visible label tokens.
- Added autotests
    - For label providers: verify token naming, and that label provider attribute updates result in token updates
- Updated [Storybook](https://60e89457a987cf003efc0a5b-ohnshtynly.chromatic.com/)
   - Rich Text Editor has a new "Localizable labels" row in their Storybook controls/API section
   - Under `Tokens/Label Providers` is a new  `Rich Text Editor Label Provider`.

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts
Comment (vikisekarNI) [inline packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts:1]: Since this utility is used in testing, should it go under `testing` folder?

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (vikisekarNI) [inline packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts:79]: await: Should we use `getSpecTypeByNamedList` pattern here to check for all formatting buttons?

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (vikisekarNI) [inline packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts:45]: nit: To represent more that this string is customized app specific label string, can we have `Customized Bold Label`

Consider applying for all instances.

Comment (aagash-ni) [reply packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts]: Updated

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (aagash-ni) [reply packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts]: Updated test cases with  `getSpecTypeByNamedList`

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (vikisekarNI) [inline packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts:18]: This should also go into `../testing.types.ts`?

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.stories.ts
Comment (vikisekarNI) [inline packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.stories.ts:1]: Do we need to update [`label-providers.mdx`](https://github.com/ni/nimble/pull/1328/files#diff-b240e51b1ed36cc5b8c6f4bcd4fd6b1d22dd5006bc1c86618a132215cc2b2c3c)? 

Currently it's missing the documentation update for rich text editor label providers

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (aagash-ni) [reply packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts]: Updated

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.stories.ts
Comment (aagash-ni) [reply packages/nimble-components/src/rich-text-editor/tests/rich-text-editor.stories.ts:1]: Updated

**Review by vikisekarNI — APPROVED** [review]

File: packages/nimble-components/src/label-provider/base/tests/label-providers.mdx
Comment (m-akinc) [inline packages/nimble-components/src/label-provider/base/tests/label-providers.mdx:33]: Let's put the Rich Text Editor before the Table (alphabetical order). That will match the ordering in the tree view, too.

**Review by m-akinc — APPROVED** [review]

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (aagash-ni) [reply packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts:84]: Removed the suppression at the top, But inside the loop, it throws lint error as contains unsafe references to the variable

File: packages/nimble-components/src/label-provider/base/tests/label-providers.mdx
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/base/tests/label-providers.mdx]: Updated

File: packages/nimble-components/src/label-provider/rich-text-editor/label-token-defaults.ts
Comment (jattasNI) [inline packages/nimble-components/src/label-provider/rich-text-editor/label-token-defaults.ts:10]: I checked a couple other products like GitHub and AzDO and they seem to prefer this language, which aligns better with "Numbered List".

Ideally we'd also update the names of the properties to match.

```suggestion
    richTextEditorToggleBulletListLabel: 'Bulleted List',
```

File: packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts
Comment (jattasNI) [inline packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts:11]: Rather than duplicating this logic from the table, could you pass the prefix in as a parameter and move the utility to a shared location like `label-provider/base/tests`?

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (jattasNI) [inline packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts:92]: We should not expose page object methods that return element types like `getFormattingButton`. Instead there should be dedicated page object methods that return strings for the text content and the title of a given button.

**Review by jattasNI — APPROVED** [review]
I added @msmithNI as an optional reviewer since he created the label provider infrastructure. But you don't need to wait on his approval. Now that Milan's back in office, you should wait for his review though.

File: packages/nimble-components/CONTRIBUTING.md
Comment (rajsite) [inline packages/nimble-components/CONTRIBUTING.md:452]:
```suggestion
The current label providers:

-   `nimble-label-provider-core`: Used for labels for all components without a dedicated label provider
-   `nimble-label-provider-table`: Used for labels for the table (and table sub-components / column types)
-   `nimble-label-provider-rich-text-editor`: Used for labels for the rich text editor
```

Comment (rajsite) [reply packages/nimble-components/CONTRIBUTING.md:452]: The components README.md has similar language to update:

https://github.com/ni/nimble/blob/db5ebd42c334810bb29be9c86cee763da487b39b/packages/nimble-components/README.md#L103-L106

File: packages/nimble-components/src/label-provider/base/tests/label-providers.mdx
Comment (rajsite) [inline packages/nimble-components/src/label-provider/base/tests/label-providers.mdx:24]:
```suggestion
Provides labels for all Nimble components that do not have a dedicated label provider.
```

File: packages/nimble-components/src/label-provider/rich-text-editor/label-token-defaults.ts
Comment (rajsite) [reply packages/nimble-components/src/label-provider/rich-text-editor/label-token-defaults.ts:10]: Please do it uniformly for all these properties, token names, element attributes, etc, not just for the string as shown here.

```suggestion
    richTextEditorToggleBulletedListLabel: 'Bulleted List',
```

File: packages/nimble-components/CONTRIBUTING.md
Comment (rajsite) [inline packages/nimble-components/CONTRIBUTING.md:452]: Similar to how the `nimble-label-provider-table` is used for all table related features we should name this provider `nimble-label-provider-rich-text` to support all possible rich-text related features current (`rich-text-editor` and `rich-text-viewer`) and potentially future.

We should propagate that consistently across this PR in all names, comments, etc. ie `nimble-label-provider-rich-text`

File: packages/nimble-components/src/label-provider/rich-text-editor/label-tokens.ts
Comment (rajsite) [inline packages/nimble-components/src/label-provider/rich-text-editor/label-tokens.ts:3]: Looking at the [localization token naming rules](https://github.com/ni/nimble/blob/d3969e869fd98721c65126c58543c0de3c9c1100/packages/nimble-components/CONTRIBUTING.md#localization)

I'd say these are 
element/types: rich-text
part/category: optional and omit, don't think these are ambiguous about what they are associated with or likely to conflict with future changes in rich text
functionality/sub-part: toggle-bold, toggle-italic, etc

So I'd vote for the names:
```
rich-text-toggle-bold-label
rich-text-toggle-italics-label
rich-text-toggle-bulleted-list-label
rich-text-toggle-numbered-list-label
```

@jattasNI @msmithNI does that sound reasonable?

File: packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts
Comment (rajsite) [reply packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts:11]: Ah good catch seeing it's only used in tests 👍

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (rajsite) [inline packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts:28]: Avoid referring to this as index

```suggestion
    toolbarButton: ToolbarButton
```

File: packages/nimble-components/src/label-provider/rich-text-editor/label-tokens.ts
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/rich-text-editor/label-tokens.ts]: Updated as per your suggestion

File: packages/nimble-components/CONTRIBUTING.md
Comment (aagash-ni) [reply packages/nimble-components/CONTRIBUTING.md]: Updated

File: packages/nimble-components/src/label-provider/base/tests/label-providers.mdx
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/base/tests/label-providers.mdx]: Updated

File: packages/nimble-components/CONTRIBUTING.md
Comment (aagash-ni) [reply packages/nimble-components/CONTRIBUTING.md]: Updated

File: packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts
Comment (aagash-ni) [reply packages/nimble-components/src/rich-text-editor/tests/rich-text-editor-labels.spec.ts]: Updated

File: packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/rich-text-editor/name-utils.ts]: Moved to `label-provider/base/tests/label-name-utils.ts` file and reused in 'table' and 'rich-text'

File: packages/nimble-components/src/label-provider/rich-text-editor/label-token-defaults.ts
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/rich-text-editor/label-token-defaults.ts]: Updated

File: packages/nimble-components/src/label-provider/base/tests/label-name-utils.ts
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/base/tests/label-name-utils.ts]: Updated

File: packages/nimble-components/src/label-provider/base/tests/label-provider-stories-utils.ts
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/base/tests/label-provider-stories-utils.ts]: Removed `removeNamePrefix` and added `removePrefixAndCamelCase` inline.

File: packages/nimble-components/src/label-provider/base/tests/label-name-utils.ts
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/base/tests/label-name-utils.ts:14]: Added Example for transformation

File: packages/nimble-components/src/label-provider/base/tests/label-provider-stories-utils.ts
Comment (aagash-ni) [reply packages/nimble-components/src/label-provider/base/tests/label-provider-stories-utils.ts]: Updated

**Review by msmithNI — APPROVED** [review]

**Review by rajsite — APPROVED** [review]

**Pull request merged**
